model/transaction/event.go

```diff
@@ -91,7 +91,75 @@ type SpanCount struct {
 }
 
 func DecodeRUMV3Event(input interface{}, cfg m.Config, err error) (transform.Transformable, error) {
-	return DecodeEvent(input, cfg, err)
+	transformable, err := DecodeEvent(input, cfg, err)
+	if err != nil {
+		return transformable, err
+	}
+	event, _ := transformable.(*Event)
+	raw, _ := input.(map[string]interface{})
+
+	decoder := &utility.ManualDecoder{}
+	fieldName := field.Mapper(cfg.HasShortFieldNames)
+	marks := fieldName("marks")
+
+	agent := fieldName("agent")
+	agentMarks := decoder.MapStr(raw, agent, marks)
+	agentMarks["domComplete"] = decoder.Float64Ptr(raw, fieldName("domComplete"), marks, agent)
```

Comment:
This can be made more DRY to avoid repeating every key 3 times. Something similar to
```
	decoder := &utility.ManualDecoder{}
	fieldName := field.Mapper(cfg.HasShortFieldNames)

	resolveMarks := func(m common.MapStr, key string) {
		mappedKey := fieldName(key)
		m[key] = decoder.Float64Ptr(m, mappedKey)
		delete(m, mappedKey)
	}

	marks := fieldName("marks")

	agent := fieldName("agent")
	agentMarks := decoder.MapStr(raw, agent, marks)
	resolveMarks(agentMarks, "domComplete")
	resolveMarks(agentMarks, "domInteractive")
	resolveMarks(agentMarks, "domContentLoadedEventStart")
	resolveMarks(agentMarks, "domContentLoadedEventEnd")
```